### ISSUE TITLE
Generators can be exited by a naked return

### ISSUE THREAD

**Issue by kokes** [Microsoft/pyright#15]
Hi, given this code

```
from typing import Iterable

def foo(n: int) -> Iterable[int]:
    for j in range(n):
        if j > 13:
            return

        yield j
```

I'm getting 

```
  Expression of type 'None' cannot be assigned to return type 'Iterable[int]' (7, 13)
  Function with declared type of Iterable[int] must return value (4, 20)
```

But as per [PEP-479](https://www.python.org/dev/peps/pep-0479/), I can exit a generator using a `return`, without specifying any return values. So, unless I misunderstood something - when in a generator, returns should be handled differently.

(`mypy` does not trigger anything)

**Comment by erictraut** [comment]
Thanks for the bug report. Yeah, I don't have any type checking implemented yet for generators or coroutines. It's one of the next things on my to-do list. In the meantime, I can silence this false positive.

**Comment by erictraut** [comment]
Thanks for the bug report. I've silenced this false positive in version 1.0.4, which I just published.
